src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
To make your code more readable use full naming for arguments in your array methods.
```suggestion
    ? people.filter((person) => person.sex === 'f'
      && people.find((child) => child.mother === person.name))
    : people.filter((person) => person.sex === 'f');
```
or you can also use destructuring.